app/models/conditions_response/user_email_alert.rb

```diff
@@ -0,0 +1,116 @@
+# Abstract parent class for all alerts that send out emails to users
+#
+#  * knows if an alert needs to be sent today (or on _this_date_)
+#  * knows the name of the method to send to MemberMailer to send out the
+#     alert to a user
+#
+#  TODO If each class is responsible for doing 1 thing -- i.e. sending out 1
+#   kind of alert for 1 condition, then is 'condition' really needed?
+#     (If/when we have situations when 1 class needs to handle more than 1
+#      condition _and_ we must have it handled only within 1 class,
+#        then we can create the condition attribute.)
+#
+class UserEmailAlert
+
+
+  # TODO this class is an instantiation of the Condition; why pass it again as an arg?
+  def self.condition_response(_condition, config, log, this_date: DateTime.now.utc)
+
+    User.all.each do |user|
+
+      if send_alert_this_day?(config, user, this_date)
+        MemberMailer.send(mailer_method, user)
+        log.record('info', log_message(log_msg_start, user.email))
+      end
+
+    end
+
+  end
+
+
+  # method to improve readability. returns true if day_number is in config[:days]
+  #
+  # Expects config to include the :days key; returns false if it is not there
+  #
+  # If config is not a Hash, raises an Error because it really _should_ be a Hash
+  #  (it's a programming error to call this otherwise!)
+  #
+  def self.send_on_day_number?(day_number, config)
+    config.fetch(:days, false) ? config[:days].include?(day_number) : false
+  end
+
+
+  # Subclasses should define this by checking information for the user
+  # computing whatever date information is necessary to determine if
+  # an alert should be sent out today.
+  # Ex:
+  #   def self.send_alert_this_day?(config, user)
+  #     days_until = (user.membership_expire_date - Date.current).to_i
+  #     user.membership_current? &&  config[:days].include?(days_until)
+  #   end
+  #
+  #
+  # @param config [Object] - configuration information used to determine if an alert should be sent
+  # @param user [User] - the user we are checking and, if appropriate, will send the alert to
+  # @param this_date [DateTime] - (defaults to today) the date that we are using
+  #                      to check the condition.  in UTC
+  #
+  def self.send_alert_this_day?(_config, _user, _this_date = DateTime.now.utc)
```

Comment:
Yes, it is less precise.  However, we store dates and times in UTC.  Determining expiration dates, etc., would be needlessly laborious if we converted every date to a _local_ DateTime.  What matters is relative "distance" between dates, at least for payments and for membership/h-branding status.

Now, this can create an anomaly such that, say, a user's payment might expire tomorrow, and they check their membership status at 30 minutes past midnight (that is, _tomorrow_) and see that they are still a member.  If they check an hour later, they'll see that their payment has expired.  (this scenario is based on current offset in Sweden of +01).  I had thought about this when developing payments but decided it really was the better tradeoff to go for simplicity rather than relatively needless accuracy.

BTW, that scenario also assume that we have a response class that runs a little after midnight that checks membership status and resets status when the right conditions are met - we have to develop that once we get this support structure in place. (I just added a PT story for that)